float32 percent_free
